src/fuse_reduce.cpp

```diff
@@ -197,13 +197,31 @@ struct find_pointwise_reduce
         return match::name("fused_reduce")(match_broadcastable_input("pointwise", "pointwise"));
     }
 
+    bool is_valid_broadcast(const instruction_ref b, const std::vector<size_t> reduce_axes) const
+    {
+        std::vector<size_t> broadcast_indices;
+        auto bstrides = b->get_shape().strides();
+
+        return std::all_of(
+            reduce_axes.begin(), reduce_axes.end(), [&](auto a) { return bstrides.at(a) == 0; });
```

Comment:
I think you need to check if the non-reduce axes are not being broadcasted as well.